models/magiceden/solana/magiceden_solana_trades.sql

```diff
@@ -0,0 +1,243 @@
+            , 'v2' as version
+            , date_trunc('day', t.call_block_time) as block_date
+            , date_trunc('month', t.call_block_time) as block_month
+            , t.call_block_time as block_time
+            , tk.token_name
+            , tk.token_symbol
+            , 'secondary' as trade_type
+            , token_size as number_of_items --all single trades right now
+            , t.trade_category
+            , t.account_buyer as buyer
+            , t.account_seller as seller
+            , t.price as amount_raw --magiceden does not include fees in the emitted price
+            , t.price/1e9 as amount_original
+            , t.price/1e9 * sol_p.price as amount_usd
+            , 'SOL' as currency_symbol
+            , 'So11111111111111111111111111111111111111112' as currency_address
+            , t.account_metadata --token id equivalent
+            , tk.account_master_edition --token id equivalent
+            , tk.account_mint --token id equivalent
+            , tk.verified_creator --contract address equivalent
+            , tk.collection_mint --contract address equivalent
+            , 'TSWAPaqyCSx2KABk68Shruf4rp7CxcNi8hAsbdwmHbN' as project_program_id
+            , cast(null as varchar) as aggregator_name
+            , cast(null as varchar) as aggregator_address
+            , t.call_tx_id as tx_id
+            , t.call_block_slot as block_slot
+            , t.call_tx_signer as tx_signer
+            , t.taker_fee as taker_fee_amount_raw --taker fees = platform fees
+            , t.taker_fee/1e9 as taker_fee_amount
+            , t.taker_fee/1e9 * sol_p.price as taker_fee_amount_usd
+            , case when t.taker_fee = 0 then 0 else t.taker_fee/t.price end as taker_fee_percentage
+            , t.maker_fee as maker_fee_amount_raw
+            , t.maker_fee/1e9 as maker_fee_amount
+            , t.maker_fee/1e9 * sol_p.price as maker_fee_amount_usd
+            , case when t.maker_fee = 0 then 0 else t.maker_fee/t.price end as maker_fee_percentage
+            , cast(null as double) as amm_fee_amount_raw
+            , cast(null as double) as amm_fee_amount
+            , cast(null as double) as amm_fee_amount_usd
+            , cast(null as double) as amm_fee_percentage
+            , t.royalty as royalty_fee_amount_raw 
+            , t.royalty/1e9 as royalty_fee_amount
+            , t.royalty/1e9 * sol_p.price as royalty_fee_amount_usd
+            , case when t.royalty = 0 then 0 else t.royalty/t.price end as royalty_fee_percentage
+            , t.instruction
+            , t.outer_instruction_index
+            , coalesce(t.inner_instruction_index,0) as inner_instruction_index
+        FROM trades t
+        LEFT JOIN {{ ref('tokens_solana_nft') }} tk
+            ON t.account_metadata = tk.account_metadata
+        LEFT JOIN {{ source('prices', 'usd') }} sol_p ON sol_p.blockchain = 'solana' and sol_p.symbol = 'SOL' and sol_p.minute = date_trunc('minute', t.call_block_time) --get sol_price
```

Comment:
what do we think about following the new pattern set in `nft.trades`, where we remove the dependency to metadata tables like tokens/prices from the project-level spells, and focus solely on base/decoded data?

this helps with orchestration, both in prod and future CI tests, as well as when building spells downstream in the future. in the new setup, we have a universal macro that all project-level spells can then join to which does the join to tokens/prices/other metadata, `enrich_trades_macro` or something similar.

@0xRobin do you think we can apply that design here too?

keep in mind, we are going to apply similar design to `dex.trades` soon too. it is a design practice we're trying to implement across all of spellbook

Reply:
one callout:
i recognize the token metadata for solana appears to be dynamic (reads from source tables), so it is likely going to be modified less often than the manually tracked evm ones, which was the main reason for our pain point. i do still think it's a good practice to apply here though

Reply:
fair, I've removed collection_mint which is the only one that can change. the other accounts cannot change.